cid/common.py

```diff
@@ -763,30 +759,56 @@ def update(self, dashboard_id, recursive=False, force=False, **kwargs):
         return self._deploy(dashboard_id, recursive=recursive, update=True)
 
 
-    def update_dashboard(self, dashboard_id, recursive=False, **kwargs):
+    def update_dashboard(self, dashboard_id, recursive=False, required_datasets=None, dashboard_datasets=None, **kwargs):
 
         dashboard = self.qs.dashboards.get(dashboard_id)
         if not dashboard:
             print(f'Dashboard "{dashboard_id}" is not deployed')
             return
 
         print(f'\nChecking for updates...')
-        print(f'Deployed template: {dashboard.deployedTemplate.arn}')
-        print(f"Latest template: {dashboard.sourceTemplate.arn}/version/{dashboard.latest_version}")
+         
         try:
-            print(f"CID Version: {dashboard.deployedTemplate.cid_version}")
+            cid_version = dashboard.deployedTemplate.cid_version
         except ValueError:
-            print(f"CID Version: Not available")
-            pass
-        
+            cid_version = "N/A"
+            
         try:
-            version_latest = dashboard.sourceTemplate.cid_version if isinstance(dashboard.sourceTemplate, CidQsTemplate) else "UNKNOWN"
-            print(f"CID Latest Version: {version_latest}")
+            cid_version_latest = dashboard.sourceTemplate.cid_version if isinstance(dashboard.sourceTemplate, CidQsTemplate) else "UNKNOWN"
         except ValueError:
-            print(f"CID Latest Version: Not available")
-            pass
+            cid_version_latest = "N/A"
         
+        if dashboard.latest:     
+            print(f"  You are up to date!")       
+            print(f"  CID Version      {cid_version}")
+            print(f"  TemplateVersion  {dashboard.deployed_version} ")
+        else:
+            cid_version = str(cid_version).ljust(9, ' ')
+            cid_version_latest = str(cid_version_latest).ljust(6, ' ')
+            template_version = str(dashboard.deployed_version).ljust(9, ' ')
+            template_version_latest = str(dashboard.latest_version).ljust(6, ' ')
+            print(f"An update is available:")
+            print("                   Deployed -> Latest")
+            print(f"  CID Version      {cid_version}   {cid_version_latest}")
+            print(f"  TemplateVersion  {template_version}   {template_version_latest}")
+
+        
+        compatible_version = None        
+        try:
+            compatible_version = dashboard.sourceTemplate.cid_version.compatible_versions(dashboard.deployedTemplate.cid_version)
+        except ValueError as e:
+            logger.info(e)
         
+        if not compatible_version and compatible_version is not None:
```

Comment:
This will ask user if reclusive flag is provided already and also in the initial deployment?